.github/workflows/build.yml

```diff
@@ -76,15 +76,15 @@ jobs:
       - name: Run Conformance Tests
         if: github.base_ref == 'main' # only PRs onto main
         continue-on-error: true
-        uses: gradle/gradle-build-action@v2
+        uses: gradle/actions/setup-gradle@v3
```

Comment:
I think the recommended pattern is to have a single, separate step to set up Gradle, and then to just run Gradle as a normal shell invocation.

See the `publish-snapshot` job above.